application/service/address.go

```diff
@@ -58,11 +75,49 @@ func (service AddressService) GetByID(contextControl domain.ContextControl, ID i
 		return domain.AddressDomain{}, exists, nil
 	}
 	hash, _ := json.Marshal(address)
+
 	if err = service.AddressDomainCacheRepository.Set(contextControl,
 		service.getCacheKey(AddressCacheKeyTypeID, strconv.FormatInt(address.ID, 10)),
 		string(hash), AddressCacheTTL); err != nil {
 		service.LoggerSugar.Infow(AddressErrorToGetByIDInCache, "address_id", address.ID)
+		return domain.AddressDomain{}, exists, err
 	}
 
 	return address, exists, nil
 }
+
+func (service AddressService) ValidateAddress(address domain.AddressDomain) error {
+	var errors []error
+
+	if len(strings.TrimSpace(address.Street)) == 0 {
+		errors = append(errors, fmt.Errorf(StreetIsRequired))
+	}
+	if len(strings.TrimSpace(address.Number)) == 0 {
+		errors = append(errors, fmt.Errorf(NumberIsRequired))
+	}
+	if len(strings.TrimSpace(address.Neighborhood)) == 0 {
+		errors = append(errors, fmt.Errorf(NeighborhoodIsRequired))
+	}
+	if len(strings.TrimSpace(address.ZipCode)) == 0 {
+		errors = append(errors, fmt.Errorf(ZipCodeIsRequired))
+	}
+	if len(strings.TrimSpace(address.City)) == 0 {
+		errors = append(errors, fmt.Errorf(CityIsRequired))
+	}
+	if len(strings.TrimSpace(address.State)) == 0 {
+		errors = append(errors, fmt.Errorf(StateIsRequired))
+	}
+	if len(strings.TrimSpace(address.Country)) == 0 {
+		errors = append(errors, fmt.Errorf(CountryIsRequired))
+	}
+
+	if len(errors) == 0 {
+		return nil
+	}
+
+	errorMessages := make([]string, len(errors))
```

Comment:
Eu não entendi isso muito bem. Qual o motivo de se fazer uma array de error e de deixar de fazer um array de string logo de início? Daria problema ter uma lista de string logo de início?